src/ecs/commands/runCommandInContainer.ts

```diff
@@ -0,0 +1,134 @@
+            localize(
+                'AWS.command.ecs.runCommandInContainer.noTasks',
+                'There are no running tasks for the service: {0}',
+                node.serviceName
+            )
+        )
+        return
+    }
+
+    const quickPick = picker.createQuickPick({
+        options: {
+            title: 'Choose a task',
+            ignoreFocusOut: true,
+        },
+        items: quickPickItems,
+    })
+
+    let taskChoice
+
+    if (quickPickItems.length === 1) {
+        taskChoice = quickPickItems
+    } else {
+        taskChoice = await picker.promptUser({
+            picker: quickPick,
+            onDidTriggerButton: (button, resolve, reject) => {
+                if (button === vscode.QuickInputButtons.Back) {
+                    resolve(undefined)
+                }
+            },
+        })
+    }
+
+    if (!taskChoice) {
+        return
+    }
+
+    const task = taskChoice[0].label
+
+    //Get command line text from user to run in the container
+    const command = await window.showInputBox({
+        prompt: localize('AWS.command.ecs.runCommandInContainer.prompt', 'Enter the command to run in container'),
+        placeHolder: localize('AWS.command.ecs.runCommandInContainer.placeHolder', 'Command to run'),
+        ignoreFocusOut: true,
+    })
+    if (!command) {
+        return
+    }
+
+    // Warn the user that commands may modify the container, give the option to dismiss future prompts
+    const configuration = new DefaultSettingsConfiguration(extensionSettingsPrefix)
+    if (configuration.readSetting('ecs.warnRunCommand')) {
+        const choice = await window.showInformationMessage(
+            localize(
+                'AWS.command.ecs.runCommandInContainer.warnBeforeExecute',
+                'Command may modify the running container {0}. Are you sure?',
+                node.continerName
+            ),
+            { modal: true },
+            localize('AWS.generic.response.yes', 'Yes'),
+            localize('AWS.command.ecs.runCommandInContainer.yesDoNotShowAgain', "Yes, and don't ask again")
+        )
+        if (choice === undefined) {
+            getLogger().debug('ecs: Cancelled runCommandInContainer')
+            return
+        } else if (
+            choice === localize('AWS.command.ecs.runCommandInContainer.yesDoNotShowAgain', "Yes, and don't ask again")
+        ) {
+            configuration.writeSetting('ecs.warnRunCommand', false, vscode.ConfigurationTarget.Global)
+        }
+    }
+
+    const response = await new ChildProcess(
+        true,
+        'aws',
+        undefined,
+        'ecs',
+        'execute-command',
+        '--cluster',
+        node.clusterArn,
+        '--task',
+        task,
+        '--container',
+        node.continerName,
+        '--command',
+        command,
+        '--interactive'
+    ).run()
+
+    getLogger().info(response.stdout)
+    if (response.exitCode !== 0 || response.stderr) {
```

Comment:
presense of stderr is not an error, if exitCode is zero. Unless there is a special case for `aws` or `aws ecs execute-command`, that you are aware of (in that case, add a 1-line comment specifying this unusual case)

Reply:
no special case. corrected.